### PR TITLE
Fix tests propshaft

### PR DESCRIPTION
**Pull request by lazaronixon**
This PR fixes tests blocking Propshaft from being merged to rails.

### CODE REVIEW

File: actiontext/app/helpers/action_text/content_helper.rb [hunk 1]
```diff
@@ -2,7 +2,9 @@
 
 # :markup: markdown
 
-require "rails-html-sanitizer"
+unless defined?(Rails::HTML::Sanitizer)
+  require "rails-html-sanitizer"
+end
```
Comment (lazaronixon) [inline actiontext/app/helpers/action_text/content_helper.rb:7]: It prevents circular require warnings when executing `actiontext`, `activestorage`, and `actionbox`.

Comment (zzak) [reply actiontext/app/helpers/action_text/content_helper.rb:7]: If you can reproduce this outside of this context could you send a PR?

File: railties/test/application/assets_test.rb [hunk 2]
```diff
@@ -156,28 +148,10 @@ class User < ActiveRecord::Base; raise 'should not be reached'; end
       assert_no_file_exists("#{app_path}/public/assets/something.else-*.css")
     end
 
-    test "precompile something.js for directory containing index file" do
```
Comment (zzak) [inline railties/test/application/assets_test.rb:159]: Curious what you're planning to do with the sprockets assets tests

File: actiontext/app/helpers/action_text/content_helper.rb (hunk 1 above)
Comment (lazaronixon) [reply actiontext/app/helpers/action_text/content_helper.rb:7]: Sure, I'll do more investigation on that, too.

File: railties/test/application/assets_test.rb (hunk 2 above)
Comment (lazaronixon) [reply railties/test/application/assets_test.rb:159]: My idea is to change tests to be compatible with Propshaft and remove what is not. Most of them are related to the fact that Propshaft can't compile `.erb` files.

Comment (zzak) [reply railties/test/application/assets_test.rb:159]: I don't think we should lose the sprockets tests, since we're not deprecating/removing it afaik

**Pull request closed without merging**